src/ds5/ds5-thermal-monitor.h

```diff
@@ -0,0 +1,72 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2021 Intel Corporation. All Rights Reserved.
+#pragma once
+
+#include "sensor.h"
+#include "device-calibration.h"
+
+namespace librealsense
+{
+    class  ds5_thermal_monitor
+    {
+    public:
+        ds5_thermal_monitor(synthetic_sensor& activation_sensor, std::shared_ptr<option> temp_option);
+        ~ds5_thermal_monitor();
+
+        void update(bool on);
+        bool is_running() const { return _is_running; }
+        void add_observer(std::function<void(float)> callback)
+        {
+            _thermal_changes_callbacks.push_back(callback);
+        }
+
+    private:
+        ds5_thermal_monitor(const ds5_thermal_monitor&) = delete;       // disable copy and assignment ctors
+        ds5_thermal_monitor& operator=(const ds5_thermal_monitor&) = delete;
+
+        // Active Object's main routine
+        void start();
+        void stop();
+        void polling(dispatcher::cancellable_timer cancellable_timer);
+        void notify(float  temperature) const;
+
+        std::weak_ptr<synthetic_sensor> _dpt_sensor;
+        active_object<> _monitor;
+        unsigned int _poll_intervals_ms;
+        float _thermal_threshold_deg;
+        float _temp_base;
+        bool _is_running = false;
```

Comment:
Can't we use _monitor.is_running()? I'd rather add the API to there...

Reply:
`is_running()` is in the class' public interface

Reply:
I meant that active_object already tracks if it's running or not. Why have another member here?

Reply:
added